src/translator/batch_translator.h

```diff
@@ -12,7 +12,7 @@
 #include "translator/history.h"
 #include "translator/scorers.h"
 
-#ifdef WITH_PTHREADS
+#ifndef __EMSCRIPTEN__
```

Comment:
@jerinphilip Please use `WASM` preprocessor instead of `__EMSCRIPTEN__` for these purposes. We want to migrate away from EMSCRIPTEN tool chain and then this code will fail to compile.

In the top level CMakeLists.txt, you can define it like this:

```
if(COMPILE_WASM)
    # Add compile definition for wasm builds
    add_compile_definitions(WASM)
endif()

```

The same preprocessor directive is followed in marian sources as well.

Reply:
WASM seems to be added to libbergamot-translator already. I was worried it might not propagate, but builds seem passing so we should be good.

Reply:
@jerinphilip That is happening because we are importing compilation definitions and flags from marian. We ideally should not.

But it is my recommendation to include:

```
if(COMPILE_WASM)
    # Add compile definition for wasm builds
    add_compile_definitions(WASM)
endif()

```
explicitly in `src/translator/CMakeLists.txt` just to avoid surprises in future 👍

Reply:
No, 1) I have no means to test this change without removing the other. 2) Editing there is outside the scope of this PR. Please approve and move this to a "potential issue".

Reply:
Also, on second checks, WASM is already at [src/translator/CMakeLists.txt#L24](https://github.com/browsermt/bergamot-translator/blob/258f23edcc045758304555971afb77f046e00f6e/src/translator/CMakeLists.txt#L24), which is why the builds are passing as mentioned before.

> WASM seems to be added to libbergamot-translator already.

There's nothing to do here.